library/pk.c

```diff
@@ -853,6 +854,88 @@ mbedtls_pk_type_t mbedtls_pk_get_type(const mbedtls_pk_context *ctx)
 }
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
+int mbedtls_pk_get_ec_public_key_props(mbedtls_pk_context *pk,
+                                       psa_ecc_family_t *ec_curve, size_t *bits)
+{
+    if ((pk == NULL) || (ec_curve == NULL) || (bits == NULL)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+    if ((pk->pk_ec_family == 0) ||
+        (pk->pk_bits == 0)) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+
+    *ec_curve = pk->pk_ec_family;
+    *bits = pk->pk_bits;
+
+    return 0;
+}
+
+#if defined(MBEDTLS_ECP_LIGHT)
+int mbedtls_pk_update_public_key_from_keypair(mbedtls_pk_context *pk)
+{
+    int ret = MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE;
+    mbedtls_ecp_keypair *ecp_keypair;
+
+    if (pk == NULL) {
+        return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+    }
+    /* RSA does not support raw public keys inside the pk_context structure,
```

Comment:
The comment should be updated to match the check, along the lines of "Only the following key types are supported."

(Same in the other function if we keep it.)